client/rpcdataset.go

```diff
@@ -0,0 +1,448 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package client
+
+import (
+	"context"
+	"encoding/binary"
+	"fmt"
+	"math"
+	"strconv"
+	"time"
+
+	"github.com/apache/iotdb-client-go/rpc"
+)
+
+const (
+	TIMESTAMP_STR = "Time"
+	START_INDEX   = 2
+	FLAG          = 0x80
+)
+
+type IoTDBRpcDataSet struct {
+	columnCount                int
+	sessionId                  int64
+	queryId                    int64
+	lastReadWasNull            bool
+	rowsIndex                  int
+	queryDataSet               *rpc.TSQueryDataSet
+	sql                        string
+	fetchSize                  int32
+	columnNameList             []string
+	columnTypeList             []int32
+	columnOrdinalMap           map[string]int32
+	columnTypeDeduplicatedList []int32
+	columnNameIndexMap         map[string]int32
+	columnTypeMap              map[string]int32
```

Comment:
These two are useless, we can use columnOrdinalMap and columnTypeDeduplicatedList to infer the data type

Reply:
PTAL https://github.com/apache/iotdb-client-go/pull/3/commits/1c2a0a3d62010d8b7938560ea4fc8c83ad1ce87d